app/models/payment_method.rb

```diff
@@ -0,0 +1,24 @@
+class PaymentMethod
+  attr_reader :name, :code
+
+  def self.all
+    response = Faraday.get('paymentmethods.com/v1/api/all')
+    return [] if response.status == 403
+    return [] if response.status == 400
+    json_response = JSON.parse(response.body, symbolize_names: true)
+    payment_methods = []
+    json_response.each do |r|
+      payment_methods << PaymentMethod.new(name: r[:name], code: r[:code])
+    end
+    return payment_methods
```

Comment:
Acho que uma troca simples desse each pra um map pode evitar várias variáveis, porque aí o retorno já vai transformado, o que acha?

```suggestion
    json_response.map do |response|
      PaymentMethod.new(response) # se o json vem exatamente nesse formato talvez você consiga passar o hash direto
    end
```